libraries/Microsoft.Bot.Builder.GrammarChecker/Checker.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Bot.Builder.GrammarChecker.CorrectingInfos;
+using Microsoft.Bot.Builder.GrammarChecker.SyntaxFeatures;
+
+namespace Microsoft.Bot.Builder.GrammarChecker
+{
+    public class Checker
+    {
+        private ISyntaxModule syntaxModule;
+        private Transducer transducer;
+
+        public Checker(ISyntaxModule syntaxModule)
+        {
+            this.syntaxModule = syntaxModule;
+            this.transducer = new Transducer();
+        }
+        
+        public string CheckText(string text)
```

Comment:
personal think there could have two functions, one is 
`List<RevisedProposal> CheckText(text)`,   to get the suggestion diagnostics from a text.

This function provides users with suggestions for various modifications
the structure of RevisedProposal similars to this
```
ReviseType Type; // modify, delete, add, 
string OriginText;
int StartIndex; // if we support multi line ,a Range object is recommend
int EndIndex;
string RevisedResult;
```
for example: `I have a apple`, suggestion should be 
```
[
   {
        OriginText: 'I have a apple', 
        Type: 'modify', 
        StartIndex:7, 
        EndIndex:8,
        RevisedResult: 'an' 
   }
]
```

another function is get the most accurate `autocorrected` answer from the proposal list. like:
`string GetAutoCorrect(List<RevisedProposal> proposals)`

this function can used as One-click modification.